### PR TITLE
Support azure functions resource

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description

Add azure_web_app_function(s) resources

### Issues Resolved
https://github.com/inspec/inspec-azure/issues/373

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/azure_web_app_functions.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+class AzureWebAppFunctions < AzureGenericResources
+  name 'azure_web_app_functions'
+  desc 'Verifies settings for a collection of Azure PostgreSQL Databases on a PostgreSQL Server'
```
Comment (clintoncwolfe) [inline libraries/azure_web_app_functions.rb:3]: Bit of copy-paste here - Note the "PostgreSQL Databases"

File: test/integration/verify/controls/azure_web_app_function_test.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+control 'azure_web_app_function' do
+  describe azure_web_app_function(resource_group: resource_group, site_name: site_name, function_name: function_name) do
+    it { should exist } # The test itself.
+    its('name')                             { should cmp "#{site_name}/#{function_name}" }
+    its('type')                             { should cmp 'Microsoft.Web/sites/functions' }
+    its('properties.name')                  { should cmp function_name }
+    its('properties.language')              { should cmp 'Javascript' }
+  end
+end
```
Comment (clintoncwolfe) [inline test/integration/verify/controls/azure_web_app_function_test.rb]: 👍

File: libraries/azure_web_app_function.rb [hunk 3]
```diff
@@ -25,3 +25,26 @@ def to_s
     super(AzureWebAppFunction)
   end
 end
+
+# Provide the same functionality under the old resource name.
+# This is for backward compatibility.
+class AzurermWebAppFunction < AzureWebAppFunction
+  name 'azurerm_web_app_function'
+  desc 'Verifies settings for an Azure Virtual Network'
```
Comment (clintoncwolfe) [inline libraries/azure_web_app_function.rb:33]: Be sure to update the description.

**Review by clintoncwolfe — COMMENTED** [review]
Overall, this is great work. Clearly you've put in a lot of effort into testing and making sure that the TF code works.

There are some rough spots in the documentation that need to be smoothed out but I will leave that in the capable hands of the Docs team.

This looks great in reading, but I hesitate to approve because I am not an Azure expert and I would not want to imply that it is more solid than it actually is - you'll need to rely on your own testing for that. But it contains all the parts I would expect in a cloud resource pack new resource PR, and it is well enough done.

Good work!

File: libraries/azure_web_app_functions.rb (hunk 1 above)
Comment (sathish-progress) [reply libraries/azure_web_app_functions.rb:3]: nice catch actually!

**Comment by sathish-progress** [comment]
@IanMadd  Could you please review this one?

**Comment by IanMadd** [comment]
I made some edits to the two docs pages. Can you just verify that my edits are correct.

**Review by IanMadd — APPROVED** [review]
LGTM
tenor-86874235

**Review by ppradhan9 — APPROVED** [review]
LGTM

**Review by ppradhan9 — APPROVED** [review]
Thanks! @sathish-progress

**Review by Nirbhay1997 — APPROVED** [review]

**Pull request merged**